frontend/src/app/core/api.service.ts

```diff
@@ -75,36 +82,25 @@ export class ApiService {
         return this.presentExerciseHelper.ownClientId !== undefined;
     }
 
-    private readonly timeTravelHelper: TimeTravelHelper = new TimeTravelHelper(
-        () =>
-            this.isTimeTraveling
-                ? this.presentState!
-                : getStateSnapshot(this.store).exercise,
-        (state) => this.store.dispatch(setExerciseState(state)),
-        async () =>
-            lastValueFrom(
-                this.httpClient.get<ExerciseTimeline>(
-                    `${httpOrigin}/api/exercise/${this.exerciseId}/history`
-                )
-            ).catch((error) => {
-                this.stopTimeTravel();
-                this.messageService.postError({
-                    title: 'Die Vergangenheit konnte nicht geladen werden',
-                    error,
-                });
-                throw error;
-            })
-    );
+    private timeTravelHelper?: TimeTravelHelper;
 
-    public readonly jumpToTime = this.timeTravelHelper.jumpToTime.bind(
-        this.timeTravelHelper
-    );
-    public timeConstraints$ = this.timeTravelHelper.timeConstraints$;
-    public get timeConstraints() {
-        return this.timeTravelHelper.timeConstraints;
+    public jumpToTime(exerciseTime: number) {
+        if (!this.timeTravelHelper) {
+            this.messageService.postError({
+                title: 'Zeitreise ist nicht aktiviert.',
+            });
+        }
+        this.timeTravelHelper?.jumpToTime(exerciseTime);
     }
-    public get isTimeTraveling() {
-        return this.timeTravelHelper.isTimeTraveling;
+
+    public timeConstraints$: Observable<TimeConstraints | undefined> =
+        this.isTimeTraveling$.pipe(
+            switchMap(
+                () => this.timeTravelHelper?.timeConstraints$ ?? of(undefined)
+            )
```

Comment:
Always when `isTimeTraveling` changes, we either emit the `timeConstraints` or `undefined` - depending on whether the timeTravelHelper is defined.

This could also be written as 
```ts
(isTimeTraveling) => isTimeTraveling? this.timeTravelHelper!.timeConstraints$ : of(undefined)
```

the `of(undefined)` is necessary, because `switchMap` wants an observable.